pkg/providers/tinkerbell/create.go

```diff
@@ -19,21 +19,16 @@ func (p *Provider) BootstrapClusterOpts(_ *cluster.Spec) ([]bootstrapper.Bootstr
 	var opts []bootstrapper.BootstrapClusterOption
 
 	if p.clusterConfig.Spec.ProxyConfiguration != nil {
-		// +2 for control plane endpoint and tinkerbell IP.
-		noProxyAddresses := make([]string, 0, len(p.clusterConfig.Spec.ProxyConfiguration.NoProxy)+2)
+		var noProxyAddresses []string
 		noProxyAddresses = append(
 			noProxyAddresses,
 			p.clusterConfig.Spec.ControlPlaneConfiguration.Endpoint.Host,
 			p.datacenterConfig.Spec.TinkerbellIP,
 		)
-		noProxyAddresses = append(noProxyAddresses, p.clusterConfig.Spec.ProxyConfiguration.NoProxy...)
 
-		env := make(map[string]string, 3)
-		env["HTTP_PROXY"] = p.clusterConfig.Spec.ProxyConfiguration.HttpProxy
-		env["HTTPS_PROXY"] = p.clusterConfig.Spec.ProxyConfiguration.HttpsProxy
-		env["NO_PROXY"] = strings.Join(noProxyAddresses, ",")
-
-		opts = append(opts, bootstrapper.WithEnv(env))
+		opts, _ = common.BootstrapClusterOpts(p.clusterConfig, noProxyAddresses...)
+	} else {
+		opts, _ = common.BootstrapClusterOpts(p.clusterConfig)
```

Comment:
looks like `common.BootstrapClusterOpts()` already handles most of the logic you have here.
Let's do something similar to what vSphere provider does here https://github.com/aws/eks-anywhere/blob/main/pkg/providers/vsphere/vsphere.go#L213
Just this should be sufficient
```
    return common.BootstrapClusterOpts(p.clusterConfig, p.datacenterConfig.Spec.TinkerbellIP)
```